frontend/src/pages/journal/components/note/note.tsx

```diff
@@ -0,0 +1,179 @@
+import { sanitizeInput } from '#libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useDebounce,
+  useEffect,
+  useFormController,
+  useParams,
+} from '#libs/hooks/hooks.js';
+import { type JournalEntryGetAllItemResponseDto } from '#packages/journal/journal.js';
+import {
+  DEFAULT_NOTE_PAYLOAD,
+  SAVE_NOTE_TIMEOUT,
+} from '#pages/journal/libs/constants/constants.js';
+import {
+  convertTextToNode,
+  getTextContentFromEditorState,
+} from '#pages/journal/libs/helpers/helpers.js';
+import {
+  OnChangePlugin,
+  PlainTextPlugin,
+} from '#pages/journal/libs/plugins/plugins.js';
+import {
+  type EditorState,
+  type NoteContent,
+} from '#pages/journal/libs/types/types.js';
+import { appActions } from '#slices/app/app-notification.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
+
+import {
+  ContentEditable,
+  LexicalComposer,
+  NoteErrorBoundary,
+} from '../components.js';
+import styles from './styles.module.scss';
+
+const Note: React.FC = () => {
+  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
+    return {
+      selectedJournalEntry:
+        journal.selectedJournalEntry as JournalEntryGetAllItemResponseDto,
+    };
+  });
+
+  const { control } = useAppForm({
+    defaultValues: {
+      title: selectedJournalEntry.title,
+      text: selectedJournalEntry.text,
+    },
+    mode: 'onChange',
+  });
+
+  const { field: titleField } = useFormController({ name: 'title', control });
+  const { field: textField } = useFormController({ name: 'text', control });
+  const { value: titleValue, onChange: onTitleChange } = titleField;
+  const { value: textValue, onChange: onTextChange } = textField;
+
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
+
+  const debouncedTitleValue = useDebounce(titleValue, SAVE_NOTE_TIMEOUT);
```

Comment:
Is it possible to use here already existing debounce function, which located in shared folder?

Reply:
no, you can see this conversation [https://bsa2023.slack.com/archives/C05P563E2MN/p1695233861264939](url)